src/bun.js/bindings/BunProcess.cpp

```diff
@@ -2778,6 +2788,70 @@ JSC_DEFINE_HOST_FUNCTION(Process_functionKill,
     return JSValue::encode(jsBoolean(true));
 }
 
+JSC_DEFINE_HOST_FUNCTION(Process_functionLoadBuiltinModule, (JSGlobalObject* globalObject, CallFrame* callFrame))
+{
+    auto* zigGlobalObject = jsCast<Zig::GlobalObject*>(globalObject);
+    VM& vm = zigGlobalObject->vm();
+    auto scope = DECLARE_THROW_SCOPE(vm);
+
+
+
+    JSValue moduleName = callFrame->argument(0);
+    if (!moduleName.isString()) {
+        throwTypeError(zigGlobalObject, scope, "The argument must be a string"_s);
+        return JSValue::encode(jsUndefined());
+    }
+
+    String moduleNameString = moduleName.toWTFString(zigGlobalObject);
+
+
+    RETURN_IF_EXCEPTION(scope, { });
+
+    if (!Bun::isBuiltinModule(moduleNameString)) {
+        throwTypeError(zigGlobalObject, scope, "The module must be a built-in module name"_s);
+        return JSValue::encode(jsUndefined());
+    }
+
+    JSString* requireMapKey = JSC::jsString(vm, moduleNameString);
+
+    JSString* dirname = jsEmptyString(vm);
+    JSString* filename = requireMapKey;
+
+    JSCommonJSModule* moduleObject = JSCommonJSModule::create(
```

Comment:
We shouldn't create a new object here on each call. It shouldn't need to create one of these at all, in fact.

Reply:
Oh, we would still need to for the wrapper object version hm

Reply:
Where/how the required module should be loaded when `Bun::fetchCommonJSModule` expects a target to put the module in (and it returns the target itself at the end)?

Reply:
Oh github just loaded this message for some reason.

> Oh, we would still need to for the wrapper object version hm
Does that mean that even though the object creation here seems to be redundant, but needed to wrap the module loaded is that correct ?